### ISSUE TITLE
Fix Rafter restore procedure

### ISSUE THREAD

**Issue by michal-hudy** [kyma-project/kyma#6539]
**Description**

Implement Rafter backup/restore tests.

Notes:
 - Tests are ready in PR
 - Documentation should be updated (remove Troubleshooting guide and update restore procedure)
 - Verify restore procedure

**Reasons**

Verification if Rafter resources are restored after cluster restore.

**Comment by michal-hudy** [comment]
Blocked, because Velero doesn't respect restoring order. Probably with the upgrade to Velero 1.2 it may be respected. There is also other way to configure that but we need input from @a-thaler about a possible solutions.

**Comment by aerfio** [comment]
https://github.com/kyma-project/kyma/pull/6675

**Comment by kazydek** [comment]
>**NOTE:** Merge the PR https://github.com/kyma-project/kyma/pull/6920 updating the Backup troubleshooting docs only if there is no change in the Kyma resource restoration order by 1.10.

**Comment by m00g3n** [comment]
https://github.com/kyma-project/test-infra/issues/1982

**Comment by pPrecel** [comment]
**Introduction**
I tested a few variants of backup/restore procedure and maybe I found a solution.  

I observe that the `velero` in version `v1.2.0` has some problems with the priority of restored resources and they fixed it in version `v.1.3.0` (like some other bugs. Read more in [release notes](https://github.com/vmware-tanzu/velero/releases/tag/v1.3.0)).

How I tested it? Like in this [script](https://github.com/kyma-project/test-infra/blob/master/prow/scripts/cluster-integration/kyma-gke-backup-test.sh) but  to create the backup instead of using this [yaml](https://github.com/kyma-project/kyma/blob/master/docs/backup/assets/backup.yaml) I used this command:
```bash 
velero backup create <BACKUP_NAME> --include-cluster-resources
``` 
But in my opinion, this point is not important. I just want only to archive my backup story. ;)

One more thing. I used the manually installed `velero` in the `velero` namespace via command:
```bash
velero install --bucket <BUCKER_NAME> gcp --secret-file <SECRET_PATH> --plugins velero/velero-plugin-for-gcp:v1.0.0,eu.gcr.io/kyma-project/backup-plugins:1ac01ae1 --wait
```
 It means that the problem may be chart configuration located in `resources/backup`.

**Tested variants**
1. `velero` in version `v1.2.0` - there are some problems where restore can't resolve some `CR's`.
2. `velero` in version `v1.2.0` + patch from this [PR](https://github.com/kyma-project/test-infra/pull/1973/files) - works fine but the backup test on test-infra ends with status `PartiallyFailed ` and I can't resolve this.
3. `velero` v1.3.0 - works fine, without errors and warnings and without the patch on deployment.

I'm not really sure but I think that the best solution for this issue should be to update the `velero` in the `kyma` and the `test-infra` to version `v1.3.0` and/or check that configuration of the chart is correct.
